File: app/src/main/java/pl/rpieja/flat/ChargesSortDialogFragment.java
package pl.rpieja.flat;

import android.app.Dialog;
import android.app.DialogFragment;
import android.os.Bundle;

public class ChargesSortDialogFragment extends DialogFragment {
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        if (!(getActivity() instanceof ChargesActivity)) return null;

        ChargesActivity activity = (ChargesActivity) getActivity();
        return new ChargesSortDialog(activity, activity.chargesViewModel)
                .create(activity.viewPager.getCurrentItem());
    }
}

File: app/src/main/java/pl/rpieja/flat/ChargesSortDialogFragment.kt
package pl.rpieja.flat

import android.app.Dialog
import android.arch.lifecycle.ViewModelProviders
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.support.v4.view.ViewPager
import pl.rpieja.flat.viewmodels.ChargesViewModel

class ChargesSortDialogFragment: DialogFragment() {
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val chargesViewModel = ViewModelProviders.of(activity!!).get(ChargesViewModel::class.java)
        val currentItem = activity!!.findViewById<ViewPager>(R.id.container).currentItem
        return ChargesSortDialog(context!!, chargesViewModel).create(currentItem)!!
    }
}